platform/jvm/capture/src/main/kotlin/io/bitdrift/capture/LoggerImpl.kt

```diff
@@ -241,19 +242,9 @@ internal class LoggerImpl(
                     ),
                 )
 
-                appExitLogger =
-                    AppExitLogger(
-                        logger = this,
-                        activityManager,
-                        runtime,
-                        errorHandler,
-                        memoryMetricsProvider = memoryMetricsProvider,
-                    )
-
-                // Install the app exit logger before the Capture logger is started to ensure
-                // that logs emitted during the installation are the first logs emitted by the
-                // Capture logger.
-                appExitLogger.installAppExitLogger()
+                commonBackgroundDispatcher.launchAsync {
+                    setupAppExitInfo(commonBackgroundDispatcher)
```

Comment:
I would just move the `appExitLogger.installAppExitLogger()` to the bg, there's not benefit of moving the construction plus it creates potential race conditions of it being null